control/veda_val.py

```diff
@@ -356,9 +356,11 @@ def send_200(self, val_api_return):
         VAL will not allow duped studio urls to be sent,
         so we must scrub the data
         """
-        for c in val_api_return['courses']:
-            if c in self.val_data['courses']:
-                self.val_data['courses'].remove(c)
+        for course in val_api_return['courses']:
+            for course_id in course.keys():
```

Comment:
its not necessary to loop on course.keys(), see the source: https://github.com/edx/edx-val/blob/141f888462f7a93c01a69ab1340f7e4481e0c085/edxval/serializers.py#L95